server/models/visit.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var q = require('q');
+var uuid = require('uuid');
+
+var postgres = require('../db/postgres');
+var elasticsearch = require('../db/elasticsearch');
+var log = require('../logger')('server.models.visit');
+
+var _verbose = function() {
+  var logline = [].join.call(arguments, ', ');
+  log.verbose(logline);
+};
+
+var visit = {
+  _onFulfilled: function _onFulfilled(msg, callback, results) {
+    _verbose(msg);
+    callback(null, results);
+  },
+  _onRejected: function _onRejected(msg, callback, err) {
+    log.warn(msg);
+    callback(err);
+  },
+  _transform: function _transform(results) {
```

Comment:
I'm so pissed that we have this crap in the model. This belongs in a view object. One way to connect them: set the fields on the model, then fire an updated event so that the view can update itself. A duller but more direct way: controller passes model into view then calls view.render, or something, orchestrating it explicitly.